src/identity-providers/add/DescriptorSettings.tsx

```diff
@@ -0,0 +1,428 @@
+                key={3}
+                data-testid="kerberos-option"
+                value={t("kerberos")}
+              />
+              <SelectOption
+                key={4}
+                data-testid="x509-option"
+                value={t("x509")}
+              />
+              <SelectOption
+                key={5}
+                data-testid="windowsDomainQN-option"
+                value={t("windowsDomainQN")}
+              />
+              <SelectOption
+                key={6}
+                data-testid="unspecified-option"
+                value={t("unspecified")}
+              />
+            </Select>
+          )}
+        ></Controller>
+      </FormGroup>
+
+      <FormGroup
+        label={t("principalType")}
+        labelIcon={
+          <HelpItem
+            helpText={th("principalType")}
+            forLabel={t("principalType")}
+            forID="kc-principalType"
+          />
+        }
+        fieldId="kc-principalType"
+        helperTextInvalid={t("common:required")}
+      >
+        <Controller
+          name="config.principalType"
+          defaultValue={t("subjectNameId")}
+          control={control}
+          render={({ onChange, value }) => (
+            <Select
+              toggleId="kc-principalType"
+              onToggle={() =>
+                setPrincipalTypeDropdownOpen(!principalTypeDropdownOpen)
+              }
+              isOpen={principalTypeDropdownOpen}
+              onSelect={(_, value) => {
+                onChange(value as string);
+                setPrincipalTypeDropdownOpen(false);
+              }}
+              selections={value}
+              variant={SelectVariant.single}
+            >
+              <SelectOption
+                key={0}
+                data-testid="subjectNameId-option"
+                value={t("subjectNameId")}
+                isPlaceholder
+              />
+              <SelectOption
+                key={1}
+                data-testid="attributeName-option"
+                value={t("attributeName")}
+              />
+              <SelectOption
+                key={2}
+                data-testid="attributeFriendlyName-option"
+                value={t("attributeFriendlyName")}
+              />
+            </Select>
+          )}
+        ></Controller>
+      </FormGroup>
+
+      <SwitchField
+        field="config.postBindingResponse"
+        label="httpPostBindingResponse"
+        isReadOnly={readOnly}
+      />
+
+      <SwitchField
+        field="config.postBindingAuthnRequest"
+        label="httpPostBindingAuthnRequest"
+        isReadOnly={readOnly}
+      />
+
+      <SwitchField
+        field="config.postBindingLogout"
+        label="httpPostBindingLogout"
+        isReadOnly={readOnly}
+      />
+
+      <SwitchField
+        field="config.wantAuthnRequestsSigned"
+        label="wantAuthnRequestsSigned"
+        isReadOnly={readOnly}
+      />
+
+      {wantAuthnSigned === "true" && (
```

Comment:
If the aforementioned changes are implemented to make this a boolean there is no need to compare this value to a string anymore.

```suggestion
      {wantAuthnSigned && (
```

Reply:
did not implement as boolean because it is stored as string on backend